### PR TITLE
#2927: comment id model update to long instead of int

### LINKED ISSUE #2927: [BUG]: Error deserializing issue comments
### What happened?

I've begun noticing errors deserializing the comment ID number when receiving issue webhooks. The value appears to be an `int` in the models but it seems GitHub has finally crossed the threshold and needs a `long`. The log was for a comment created webhook request.

Note: This is very similar to bug #2889 fixed in 10.0.0, but on comment IDs rather than issue IDs. 

### Versions

10.0.0 but bug appears to still be present in the main branch

### Relevant log output

```shell
JSON integer 2147666618 is too large or small for an Int32. Path 'comment.id', line 1, position 4002.
   at Newtonsoft.Json.JsonTextReader.ParseReadNumber(ReadType readType, Char firstChar, Int32 initialPosition)
   at Newtonsoft.Json.JsonTextReader.ParseNumber(ReadType readType)
   at Newtonsoft.Json.JsonTextReader.ReadNumberValue(ReadType readType)
   at Newtonsoft.Json.JsonTextReader.ReadAsInt32()
   at Newtonsoft.Json.JsonReader.ReadForType(JsonContract contract, Boolean hasConverter)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Deserialize(JsonReader reader, Type objectType)
   at CenterEdgeBot.Events.ActivityPayloadParser.ParseActivityPayload(String payloadType, String payloadText) in /jenkins/workspace/DevOps_CenterEdgeBot_main/src/CenterEdgeBot/Events/ActivityPayloadParser.cs:line 43
   at CenterEdgeBot.Events.GitHubWebHookProcessor.ProcessGitHubWebHookPayloadAsync(String payloadType, String payloadText) in /jenkins/workspace/DevOps_CenterEdgeBot_main/src/CenterEdgeBot/Events/GitHubWebHookProcessor.cs:line 43
   at CenterEdgeBot.GitHubLambda.HandleGitHubWebHookAsync(APIGatewayProxyRequest apiGatewayProxyRequest, ILambdaContext lambdaContext) in /jenkins/workspace/DevOps_CenterEdgeBot_main/src/CenterEdgeBot/GitHubLambda.cs:line 71
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by arxange1**
Resolves #2927

----

### Before the change?

* GetIssueComments() leads to System.OverflowException: Value was either too large or too small for an Int32.

### After the change?

* GetIssueComments() returns comments.

### Pull request checklist
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/main/community/breaking_changes.md) to help!

- [x] Yes
- [ ] No

----

### CODE REVIEW

**Comment by Bazeloth** [comment]
If i could approve this i would. This is holding us back big time.

**Review by bucataruelena — APPROVED** [review]

**Comment by kevinobruno** [comment]
I think we also need to change [IIssueCommentsClient](https://github.com/octokit/octokit.net/blob/f8b9f7c2804ad4b2b539daae4f69c0c50bd6e93f/Octokit/Clients/IIssueCommentsClient.cs#L13) and its implementation [IssueCommentsClient](https://github.com/octokit/octokit.net/blob/f8b9f7c2804ad4b2b539daae4f69c0c50bd6e93f/Octokit/Clients/IssueCommentsClient.cs#L12)?

**Comment by arxange1** [comment]
> I think we also need to change [IIssueCommentsClient](https://github.com/octokit/octokit.net/blob/f8b9f7c2804ad4b2b539daae4f69c0c50bd6e93f/Octokit/Clients/IIssueCommentsClient.cs#L13) and its implementation [IssueCommentsClient](https://github.com/octokit/octokit.net/blob/f8b9f7c2804ad4b2b539daae4f69c0c50bd6e93f/Octokit/Clients/IssueCommentsClient.cs#L12)?

Indeed, thank you!

**Comment by ioshm** [comment]
I believe [RepositoryCommentsClient](https://github.com/octokit/octokit.net/blob/main/Octokit/Clients/RepositoryCommentsClient.cs) & its interface should also be updated?

**Comment by arxange1** [comment]
> I believe [RepositoryCommentsClient](https://github.com/octokit/octokit.net/blob/main/Octokit/Clients/RepositoryCommentsClient.cs) & its interface should also be updated?

Please check

File: Octokit/Clients/ICommitCommentReactionsClient.cs
Comment (ioshm) [inline Octokit/Clients/ICommitCommentReactionsClient.cs:1]: Looks like a few comment id/number parameters were not updated to `long` in the other methods.

File: Octokit/Clients/IIssueCommentReactionsClient.cs
Comment (ioshm) [inline Octokit/Clients/IIssueCommentReactionsClient.cs:1]: Looks like a few comment id/number parameters were not updated to `long` in the other methods.

File: Octokit/Clients/IPullRequestReviewCommentReactionsClient.cs
Comment (ioshm) [inline Octokit/Clients/IPullRequestReviewCommentReactionsClient.cs:1]: Looks like a few comment id/number parameters were not updated to `long` in the other methods.

File: Octokit/Clients/IPullRequestReviewCommentReactionsClient.cs
Comment (ioshm) [inline Octokit/Clients/IPullRequestReviewCommentReactionsClient.cs:75]: Documentation is wrong here?

File: Octokit/Clients/IPullRequestReviewCommentReactionsClient.cs
Comment (ioshm) [inline Octokit/Clients/IPullRequestReviewCommentReactionsClient.cs:85]: Documentation is wrong here?

File: Octokit/Clients/IRepositoryCommentsClient.cs
Comment (ioshm) [inline Octokit/Clients/IRepositoryCommentsClient.cs:21]: Not sure if we want to rename this to `commentId`. Perhaps the maintainers should chime in on this one.

File: Octokit/Clients/IRepositoryCommentsClient.cs
Comment (ioshm) [inline Octokit/Clients/IRepositoryCommentsClient.cs:1]: Looks like a few comment id/number parameters were not updated to `long` in the other methods.

File: Octokit/Clients/IRepositoryCommentsClient.cs
Comment (arxange1) [reply Octokit/Clients/IRepositoryCommentsClient.cs:21]: It is a breaking change fix anyway. I would like to see the maintainer here to reduce the id/number confusion around the code.

File: Octokit/Clients/IRepositoryCommentsClient.cs
Comment (ioshm) [reply Octokit/Clients/IRepositoryCommentsClient.cs:1]: Unless I am mistaken, `number` do not exists on comments, and only have `id`s which should be int64 by now.

Take for example this method

Definition:
https://github.com/octokit/octokit.net/blob/fe329e7b2b3503483f527a7407185a2bd81c354a/Octokit/Clients/IRepositoryCommentsClient.cs#L123-L131

Implementation:
https://github.com/octokit/octokit.net/blob/fe329e7b2b3503483f527a7407185a2bd81c354a/Octokit/Clients/RepositoryCommentsClient.cs#L209-L225

The `number` parameter still has the type `int` when it should be `long`. The docs states that the `comment_id` (in this case, this parameter is called `number`) is of type `integer` which means any double without a fractional part - `long` should be good enough.

File: Octokit/Clients/IPullRequestReviewCommentReactionsClient.cs
Comment (arxange1) [reply Octokit/Clients/IPullRequestReviewCommentReactionsClient.cs:85]: It is.

**Comment by nickfloyd** [comment]
Hey y'all,

First off, thank you for tackling this and thank you for moving forward on these updates to make things work/better - apologies for the trouble.  This was fallout from a recent change.  You are correct this issue and ones like it are why we are [moving](https://github.blog/2024-01-03-our-move-to-generated-sdks/) to a fully generated SDK.

It looks like the tests need to be updated as well before we can ✅ and get this merged in and shipped.  

To answer the question regarding scope of change right now the only ID which cascades across a number of SDK APIs is `comment_id`, which in some cases is simply named `number` as a parameter.  Changing the name of this parameter and it's type is the right move here. Especially since this will be releases as a breaking change.

Side note: The newer, generated .NET SDK has already accounted for [this](https://github.com/octokit/dotnet-sdk/blob/6ded6b3de7583197ea11eb4089de4683008c3917/src/GitHub/Models/IssueComment.cs#L51).  That SDK gets generated nightly against the latest [OpenAPI description from the GitHub REST API.](https://github.com/github/rest-api-description/tree/main/descriptions)

**Comment by BrianCArnold** [comment]
It looks like there's a lot more places where `int number` should actually be `long commentId`, I'm going to see if I can patch all of them, looks like a bit of work.

**Comment by BrianCArnold** [comment]
@arxange1 @nickfloyd 

Take a look at https://github.com/BrianCArnold/octokit.net/tree/bugfix/%232927 
I fixed what I think are the remainder of all the Int32 commentId and reactionId issues, and all the tests should be passing now.

Just cherry-pick that commit onto this branch if it looks good, I'm trying to get my bot working again.

**Comment by Bazeloth** [comment]
> > If i could approve this i would. This is holding us back big time.
> > Edit: What the hell i can? I'll patiently wait for an official approval though.
> > image
> 
> Hey @Bazeloth While you can "approve" - because we want/need community involvement and input the merge criteria would not be met given that requires an approval from a core maintainer/owner. We do take those ☑️ checks seriously as we are going through approval though ❤️ .

Thanks that explains a lot. I was just wondering, since in our repo people that can approve usually can merge as well. Good to see you guys took precautions 👍

File: Octokit/Clients/IPullRequestReviewCommentReactionsClient.cs
Comment (arxange1) [reply Octokit/Clients/IPullRequestReviewCommentReactionsClient.cs]: Fixed

File: Octokit/Clients/IPullRequestReviewCommentReactionsClient.cs
Comment (arxange1) [reply Octokit/Clients/IPullRequestReviewCommentReactionsClient.cs]: Fixed

File: Octokit/Clients/IPullRequestReviewCommentReactionsClient.cs
Comment (arxange1) [reply Octokit/Clients/IPullRequestReviewCommentReactionsClient.cs:1]: Fixed

File: Octokit/Clients/IIssueCommentReactionsClient.cs
Comment (arxange1) [reply Octokit/Clients/IIssueCommentReactionsClient.cs:1]: Fixed

File: Octokit/Clients/ICommitCommentReactionsClient.cs
Comment (arxange1) [reply Octokit/Clients/ICommitCommentReactionsClient.cs:1]: Fixed

File: Octokit/Clients/IRepositoryCommentsClient.cs
Comment (arxange1) [reply Octokit/Clients/IRepositoryCommentsClient.cs:1]: Fixed

**Comment by BrianCArnold** [comment]
... did you copy my changes into your own commit, or just do (virtually) the same thing?

**Comment by arxange1** [comment]
> ... did you copy my changes into your own commit, or just do (virtually) the same thing?

I made the changes independently and didn't see your message before doing so.

**Comment by BrianCArnold** [comment]
Ahh, gotcha, yeah, I looked through and I see that as well. There's actually several fixes that I made that look like they got missed, let me see if I can make a patch to fix them as well.

**Review by dcairney — APPROVED** [review]

**Comment by BrianCArnold** [comment]
(Also, you fixed several things I missed as well)

**Comment by BrianCArnold** [comment]
If we're doing a breaking change on this release to avoid confusions on parameter names, such as replacing "`number`" with "`commentId`", I'd like to include fixing as many of them as possible. 

Should we include them on this PR?

(https://github.com/BrianCArnold/octokit.net/tree/FixParameterNames)

**Comment by nickfloyd** [comment]
> Should we include them on this PR?

Normally I'd be a little more risk adverse, but given the scope of change here (an id field, the no longer matches the backing type + breaking change - I can definitely get behind a little bigger changeset to resolve these issues in one go.

**Comment by nickfloyd** [comment]
I'd like to land this as soon as possible. It looks like @arxange1 had pulled in a few changes from other sources.  I'll give it another day here but after that we'll accept more PRs to address this and at least get these fixes shipped out on Monday unless anyone has any issues where this should be held up.

**Comment by arxange1** [comment]
> If we're doing a breaking change on this release to avoid confusions on parameter names, such as replacing "`number`" with "`commentId`", I'd like to include fixing as many of them as possible.
> 
> Should we include them on this PR?
> 
> (https://github.com/BrianCArnold/octokit.net/tree/FixParameterNames)

Done

**Comment by jessehouwing** [comment]
For those in here need, like our team, you can find a patched build here for now.

We'll unpublish when the new version ships.

https://github.com/orgs/globaldevopsexperience/packages/nuget/package/Octokit

**Comment by jeny-amatya-qed** [comment]
Any ETA on the new version with this fix?

**Review by nickfloyd — APPROVED** [review]

**Comment by jessehouwing** [comment]
This applies to other models as well I found today. My guess is that any `int id` should be a `long id`.

See #2931

**Pull request merged**
